.babelrc

```diff
@@ -1,23 +1,36 @@
 {
   "presets": [
-    ["@babel/preset-react", {
-      "pragma": "createElement"
-    }],
+    [
+      "@babel/preset-react",
+      {
+        "runtime": "automatic",
+        "importSource": "preact"
+      }
+    ],
 
     // Compile JS for browser targets set by `browserslist` key in package.json.
-    ["@babel/preset-env", {
-       "bugfixes": true
-    }]
+    [
+      "@babel/preset-env",
+      {
+        "bugfixes": true
+      }
+    ]
   ],
   "plugins": ["inject-args"],
   "ignore": ["**/vendor/*"],
   "env": {
     "development": {
       "presets": [
-        ["@babel/preset-react", {
-          "development": true,
-          "pragma": "createElement"
-        }]
+        [
+          "@babel/preset-react",
+          {
+            "development": true,
+            "runtime": "automatic",
+            // 2021-01-04, re-asses in the future if we can import 'preact'
+            // See: https://github.com/preactjs/preact/issues/2974
+            "importSource": "preact/compat"
```

Comment:
Hmm, does this actually import the `preact/compat` package specifically or some submodule? Importing `preact/compat` has a bunch of global side effects that affect processing of all JSX that we want to avoid.

The source code of `preact/compat/jsx-dev-runtime.js` on the other hand is just:

```js
module.exports = require('preact/jsx-runtime');
```

And that's fine, although it would be clearer what was going on if we could just tell Babel to use `preact/jsx-runtime` directly.

Reply:
It only loads `preact/compat/jsx-dev-runtime.js`, nothing else.

We need to use `preact/compat` work around because browserify has a resolution problem. Preact uses `exports` in the package.json but it seems that browserify only loads modules if the directory structure exists in the package.

Reply:
The Preact team suggested me to file an issue with Browserify regarding not able to find the `preact/jsx-dev-runtime` module: https://github.com/browserify/browserify/issues/1995

Reply:
Ah, that would make sense. The `exports` field is a relatively recent addition to Node and I don't think Browserify understands it at all. Along with native support for ES modules, this is another reason to consider other bundlers in future.

Reply:
Can you change the comment here to something like:

```js
// Use `preact/compat/jsx-dev-runtime` which is an alias for `preact/jsx-runtime`.
// See https://github.com/preactjs/preact/issues/2974.
```

Can you also add the same comment in the Babel config file for the `frontend-shared` package.